parlai/core/torch_ranker_agent.py

```diff
@@ -663,7 +709,19 @@ def _build_candidates(self, batch, source, mode):
                     label_vec_pad[0 : label_vec.size(0)] = label_vec
                     label_inds[batch_idx] = self._find_match(cand_vecs, label_vec_pad)
                     if label_inds[batch_idx] == -1:
-                        bad_batch = True
+                        if (
+                            self.opt.get('add_label_to_fixed_cands')
+                            and not self.is_training
+                        ):
+                            cand_vecs = torch.cat(
+                                (cand_vecs, label_vec_pad.unsqueeze(0))
+                            )
+                            cands.append(batch.labels[batch_idx])
+                            label_inds[batch_idx] = self._find_match(
```

Comment:
i don't think we need to `_find_match` here, this seems like wasted computation as we know exactly which index the label should be at

Reply:
yeah good call, i'll change that